torchvision/models/maxvit.py

```diff
@@ -426,7 +426,7 @@ def __init__(
     ) -> None:
         super().__init__()
 
-        layers: OrderedDict[str, Any] = OrderedDict()  # type: ignore
+        layers: Dict[str, Any] = {}
```

Comment:
This dictionary was intended to maintain the order of layers so that the sequential call at like 469 will execute them in the right order.

Reply:
@datumbox  all dicts preserve insertion order as of  Python 3.6: https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6

I was assuming this would fulfil the requirements you noted for the sequential call. Am I missing something important?

Reply:
> Since Dict can in principle be any implementation of the data structure

No, it's not an implementation detail anymore in 3.7. It's a guaranteed language feature (see link above). We can rely on it.

As noted below `Sequential` should be able to handle dicts on top of OrderedDicts now that 3.7 is the minimum required version. It might be a good opportunity to make that change upstream (looks like @pmeier  is looking into it)

Reply:
Yes dicts are *insertion ordered* from python 3.7. This is legit feature and used widely.